### PR TITLE
Deploy Backend-AishaAthmanLali

### PR DESCRIPTION
**Pull request by aishaathmanlali**
## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

Write a summary of your change. Explain why you have made it.

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

**Comment by soonick** [comment]
How are new versions deployed?
How can I access the backend?

from @aishaathmanlali I have checked and it seems that it will be an auto-deploy so the new versions should automatically be deployed on render. I hope I have understood your question and answered according to it.

File: .eslintrc.json [hunk 1]
```diff
@@ -8,6 +8,8 @@
 	"root": true,
 	"rules": {
 		"indent": "off",
-		"operator-linebreak": "off"
+		"operator-linebreak": "off",
+		"no-unused-vars": "off" 
 	}
+
```
Comment (soonick) [inline .eslintrc.json]: Delete this line

File: .eslintrc.json [hunk 2]
```diff
@@ -8,6 +8,8 @@
 	"root": true,
 	"rules": {
 		"indent": "off",
-		"operator-linebreak": "off"
+		"operator-linebreak": "off",
+		"no-unused-vars": "off"
```
Comment (soonick) [inline .eslintrc.json]: Why did you add this? We shouldn't have unused variables

File: .expo/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+> What do the files contain?
+
+- "devices.json": contains information about devices that have recently opened this project. This is used to populate the "Development sessions" list in your development builds.
+- "packager-info.json": contains port numbers and process PIDs that are used to serve the application to the mobile device/simulator.
+- "settings.json": contains the server configuration that is used to serve the application manifest.
+
+> Should I commit the ".expo" folder?
+
+No, you should not share the ".expo" folder. It does not contain any information that is relevant for other developers working on the project, it is specific to your machine.
```
Comment (soonick) [inline .expo/README.md]: It says here that you shouldn't commit this folder

File: server/app.js [hunk 4]
```diff
@@ -19,11 +17,13 @@ app.use(configuredHelmet());
...
-	app.enable("trust proxy");
-	app.use(httpsOnly());
+  app.enable("trust proxy");
+  app.use(httpsOnly());
 }
 
-app.use(apiRoot, apiRouter);
+// Comment out or remove the API router for now
+// app.use(apiRoot, apiRouter);
+
```
Comment (soonick) [inline server/app.js]: Remove

File: server/server.js [hunk 5]
```diff
@@ -1,16 +1,15 @@
 import http from "node:http";
-
-import app from "./app";
+import app from "./app"; // Keeping this for future use
```
Comment (soonick) [inline server/server.js]: Remove the comment

File: server/server.js [hunk 6]
```diff
@@ -1,16 +1,15 @@
...
 import logger from "./utils/logger";
 
 const server = http.createServer(app);
 
 server.on("listening", () => {
-	const addr = server.address();
-	const bind = typeof addr === "string" ? `pipe ${addr}` : `port ${addr.port}`;
-	logger.info("listening on: %s", bind);
+  const addr = server.address();
+  const bind = typeof addr === "string" ? `pipe ${addr}` : `port ${addr.port}`;
```
Comment (soonick) [inline server/server.js]: In which case will `addr` be a string?

**Review by soonick — CHANGES_REQUESTED** [review]
Please also answer then questions I left earlier in the PR.

File: .eslintrc.json (hunk 2 above)
Comment (aishaathmanlali) [reply .eslintrc.json]: okay I will remove that

File: .expo/README.md (hunk 3 above)
Comment (aishaathmanlali) [reply .expo/README.md]: I have added it to gitignore

**Comment by aishaathmanlali** [comment]
> How are new versions deployed? How can I access the backend?

https://love-me-tender-51qa.onrender.com/

File: .expo/README.md (hunk 3 above)
Comment (soonick) [reply .expo/README.md]: The files are still appearing in your PR. You might need to make git forget them. This might work:

```
git rm --cached .expo/
```

File: netlify/functions/hello.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+// netlify/functions/hello.js
+
```
Comment (soonick) [inline netlify/functions/hello.js]: Delete

File: server/functions/app.mjs  [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import express from "express";
```
Comment (soonick) [inline server/functions/app.mjs ]: Why is the extension of this file `mjs`?

**Review by soonick — CHANGES_REQUESTED** [review]

File: .expo/README.md (hunk 3 above)
Comment (aishaathmanlali) [reply .expo/README.md]: thank you. I will try this

File: .gitignore [hunk 9]
```diff
@@ -3,3 +3,4 @@ dist/
 node_modules/
 .DS_Store
 .env
+.expo
```
Comment (soonick) [inline .gitignore]: We will not be using expo, right? We can remove this

File: client/src/pages/Home.js [hunk 10]
```diff
@@ -22,16 +21,11 @@ export function Home() {
 				console.error(err);
 			});
 	}, []);
+
```
Comment (soonick) [inline client/src/pages/Home.js]: Remove

File: package.json [hunk 11]
```diff
@@ -53,6 +53,7 @@
 		"eslint-plugin-jsx-a11y": "^6.7.1",
 		"eslint-plugin-react": "^7.33.2",
 		"eslint-plugin-react-hooks": "^4.6.0",
+		"expo-cli": "^4.13.0",
```
Comment (soonick) [inline package.json]: We don't need expo cli, do we?

File: package.json [hunk 12]
```diff
@@ -74,10 +75,11 @@
 	"dependencies": {
 		"@babel/runtime": "^7.22.15",
 		"dotenv": "^16.3.1",
-		"express": "^4.18.2",
+		"express": "^4.19.2",
 		"helmet": "^7.0.0",
 		"morgan": "^1.10.0",
 		"pg": "^8.11.3",
+		"serverless-http": "^3.2.0",
```
Comment (soonick) [inline package.json]: Do we need this?

File: server/app.js [hunk 13]
```diff
@@ -1,13 +1,12 @@
...
-
 import apiRouter from "./api";
 import config from "./utils/config";
 import {
-	clientRouter,
-	configuredHelmet,
-	configuredMorgan,
-	httpsOnly,
-	logErrors,
+  clientRouter,
```
Comment (soonick) [inline server/app.js]: It seems like you didn't change anything on this file, remove this change from the commit

File: server/functions/app.mjs  (hunk 8 above)
Comment (soonick) [inline server/functions/app.mjs ]: This file is not needed, right?

File: server/server.js [hunk 14]
```diff
@@ -1,18 +1,17 @@
 import http from "node:http";
-
 import app from "./app";
-import { connectDb, disconnectDb } from "./db";
+// import { connectDb, disconnectDb } from "./db";
```
Comment (soonick) [inline server/server.js]: Delete

**Review by soonick — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 9 above)
Comment (aishaathmanlali) [reply .gitignore]: I did remove it. But I will double check again

File: package.json (hunk 12 above)
Comment (aishaathmanlali) [reply package.json]: do we need which line? I did not understand

File: server/app.js (hunk 13 above)
Comment (aishaathmanlali) [reply server/app.js]: how do I do that

File: server/functions/app.mjs  (hunk 8 above)
Comment (aishaathmanlali) [reply server/functions/app.mjs ]: we will not use it so I have deleted the file

Comment (aishaathmanlali) [reply server/functions/app.mjs ]: yes I have deleted it

File: server/server.js (hunk 14 above)
Comment (aishaathmanlali) [reply server/server.js]: done

File: server/server.js (hunk 6 above)
Comment (aishaathmanlali) [reply server/server.js]: I am not sure about this. This was there in the starter kit but I will look into it

File: README.md [hunk 15]
```diff
@@ -16,22 +14,11 @@
...
-
-Pick one member of the team to own the repository and pipeline. That person should do the following:
-
-1.  Click the "Use this template" button above (see [GitHub's docs][1]) to create your team repository and name it something appropriate for your project.
-    - Your repo should say _"generated from"_, **not** _"forked from"_, _"CodeYourFuture/cyf-final-project-starter-kit"_ at the top
-2.  In your repo, click the "Deploy to Render" button at the top of the README and log in using GitHub when prompted.
-3.  Fill in a service group name for your application and then click "Apply".
-4.  Once it has deployed successfully, click the "managed resources" link to view the application details.
+## Deployed Website on render
+This is our deployed website link https://love-me-tender-51qa.onrender.com
```
Comment (soonick) [inline README.md:19]: Leave empty line before this line

File: README.md [hunk 16]
```diff
@@ -16,22 +14,11 @@
...
 
-Whenever you commit to main (or e.g. merge a [pull request]) it will get automatically deployed!
+The render is on auto-deploy so it should render new changes to the website.
 
-You should now make sure all of the project team are [collaborators] on the repository.
```
Comment (soonick) [inline README.md]: Leave just one empty line

File: package.json (hunk 12 above)
Comment (soonick) [reply package.json]: You added a dependency to `serverless-http`. Do wee need it?

File: server/server.js (hunk 6 above)
Comment (aishaathmanlali) [reply server/server.js]: typeof addr === "string" checks if address is a string, which would indicate a Unix domain socket or named pipe.
If address is an object (i.e., it's bound to an IP address and port), the bind variable will be set to port ${addr.port}.
This way, the code correctly logs the binding information, whether it's a network address (port) or a Unix domain socket (pipe).

File: server/app.js (hunk 13 above)
Comment (soonick) [reply server/app.js]: I think this should revert the file to the same as master:

```
git checkout master server/app.js
```

Then you can just commit the file again

File: server/server.js (hunk 6 above)
Comment (soonick) [reply server/server.js]: Ok, you can leave it like this for now

**Review by soonick — CHANGES_REQUESTED** [review]
Almost done, just some very minor things left. Thanks!

File: server/server.js (hunk 6 above)
Comment (soonick) [reply server/server.js]: We never use a unix socket or pipe, so it will never be a string

File: README.md (hunk 15 above)
Comment (aishaathmanlali) [reply README.md:19]: done

File: README.md (hunk 16 above)
Comment (aishaathmanlali) [reply README.md]: done

File: package.json (hunk 12 above)
Comment (aishaathmanlali) [reply package.json]: not anymore. I will get rid of it

File: server/app.js (hunk 13 above)
Comment (aishaathmanlali) [reply server/app.js]: done

File: server/server.js (hunk 6 above)
Comment (aishaathmanlali) [reply server/server.js]: I have changed it

**Review by soonick — APPROVED** [review]
